File: CS/16_Print/AutomaticRotationPrinting.cs
using Spire.Pdf;
using Spire.Pdf.Print;
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Drawing.Printing;
using System.Text;
using System.Windows.Forms;

namespace AutomaticRotationPrinting
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        private void button1_Click(object sender, EventArgs e)
        {
            // Create a new PDF document
            PdfDocument doc = new PdfDocument();

            // Load an existing PDF file into the document
            doc.LoadFromFile(@"..\..\..\..\..\..\Data\MultiPage.pdf");
            //Set up multi page printing layout
            PdfMultiPageLayout printParameters = doc.PrintSettings.SelectMultiPageLayout(1, 2);
            //Control whether the page automatically rotates to fit the print layout
            printParameters.AutoRotatePages = true;
            //Horizontal flipping double-sided printing mode
            doc.PrintSettings.Duplex = Duplex.Horizontal;
            // Print the document using the specified settings
            doc.Print();
            
        }
    }
}

File: VB/16_Print/AutomaticRotationPrinting.vb
Imports Spire.Pdf
Imports Spire.Pdf.Print
Imports System.ComponentModel
Imports System.Drawing.Printing
Imports System.Text

Namespace AutomaticRotationPrinting
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub button1_Click(ByVal sender As Object, ByVal e As EventArgs) Handles button1.Click
			' Create a new PDF document
			Dim doc As New PdfDocument()

			' Load an existing PDF file into the document
			doc.LoadFromFile("..\..\..\..\..\..\Data\MultiPage.pdf")
			'Set up multi page printing layout
			Dim printParameters As PdfMultiPageLayout = doc.PrintSettings.SelectMultiPageLayout(1, 2)
			'Control whether the page automatically rotates to fit the print layout
			printParameters.AutoRotatePages = True
			'Horizontal flipping double-sided printing mode
			doc.PrintSettings.Duplex = Duplex.Horizontal
			' Print the document using the specified settings
			doc.Print()

		End Sub
	End Class
End Namespace
